### PR TITLE
[EasyBatch] Add kernel.reset tag to BatchProcessor

### PR DESCRIPTION
**Pull request by alexndlm**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: packages/EasyBatch/src/BatchObjectManager.php [hunk 1]
```diff
@@ -73,10 +72,6 @@ public function approve(BatchObjectInterface $batchObject): BatchObjectInterface
                 $this->approve($this->batchItemRepository->findOrFail($batchObject->getParentBatchItemId()));
             }
 
-            // Approving a batch is possible only when all its batchItems succeeded and approval was required
```
Comment (alexndlm) [inline packages/EasyBatch/src/BatchObjectManager.php:76]: We have dispatch event next in a `\EonX\EasyBatch\Processors\BatchProcessor::dispatchBatchEvent`

Comment (natepage) [reply packages/EasyBatch/src/BatchObjectManager.php:76]: @alexndlm this event dispatch is to handle the case when we are approving the batch itself which in this case we don't go through the batch processor, or maybe I don't understand the issue.
Could you please explain how to reproduce the bug? Thanks

**Review by natepage — CHANGES_REQUESTED** [review]

File: packages/EasyBatch/src/BatchObjectManager.php (hunk 1 above)
Comment (alexndlm) [reply packages/EasyBatch/src/BatchObjectManager.php]: I have rollbacked this. I will research this issue in addition.

**Review by akl-95 — APPROVED** [review]

**Review by akl-95 — APPROVED** [review]

**Comment by natepage** [comment]
@alexndlm I'm not convinced with this solution. It feels really obscure like this, I think it should be more explicit that we need to reset services before processing a batchItem and therefore I think it should be in the Messenger middleware.

**Comment by alexndlm** [comment]
@natepage I have refactored. Please check.

**Comment by natepage** [comment]
@alexndlm I prefer the solution from #969 
what do you think?

**Comment by alexndlm** [comment]
> @alexndlm I prefer the solution from #969
> what do you think?

@natepage This will not now work.
We need to clear the cache after the item is processing.

After $this->processBatchItemWithLock (line 57) we will have items in cache.

**Comment by natepage** [comment]
Thanks @alexndlm for this.
Was done in #969

**Pull request closed without merging**